homeassistant/components/intent_script/__init__.py

```diff
@@ -111,22 +112,41 @@ async def _handle_reload(servie_call: ServiceCall) -> None:
     return True
 
 
+class _IntentSpeechRepromptData(TypedDict, total=False):
+    """Intent config data type for speech or reprompt info."""
+
+    content: template.Template
+    title: template.Template
+    text: template.Template
+    type: str
+
+
+class _IntentCardData(TypedDict, total=False):
+    """Intent config data type for card info."""
+
+    type: str
+    title: template.Template
+    content: template.Template
+
+
 class ScriptIntentHandler(intent.IntentHandler):
     """Respond to an intent with a script."""
 
-    def __init__(self, intent_type, config):
+    def __init__(self, intent_type: str, config: ConfigType) -> None:
         """Initialize the script intent handler."""
         self.intent_type = intent_type
         self.config = config
 
-    async def async_handle(self, intent_obj: intent.Intent):
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
         """Handle the intent."""
-        speech = self.config.get(CONF_SPEECH)
-        reprompt = self.config.get(CONF_REPROMPT)
-        card = self.config.get(CONF_CARD)
-        action = self.config.get(CONF_ACTION)
-        is_async_action = self.config.get(CONF_ASYNC_ACTION)
-        slots = {key: value["value"] for key, value in intent_obj.slots.items()}
+        speech: _IntentSpeechRepromptData | None = self.config.get(CONF_SPEECH)
+        reprompt: _IntentSpeechRepromptData | None = self.config.get(CONF_REPROMPT)
+        card: _IntentCardData | None = self.config.get(CONF_CARD)
+        action: script.Script | None = self.config.get(CONF_ACTION)
+        is_async_action: bool = self.config[CONF_ASYNC_ACTION]
```

Comment:
async action has a default and is optional, but shouldn't this be a `.get()` as well?

Reply:
It will never be `None`.

Reply:
Yes, I'll have a look after lunch.

Reply:
The field has a boolean validator if something is configured, and `False` as default. So a valid config will will always add a boolean for `config["CONF_ASYNC_OPTION"]`

Reply:
Good to know 👍